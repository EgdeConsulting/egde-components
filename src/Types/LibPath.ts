/* eslint-disable */

export enum Root {
    Introduction = 'Getting Started/',
    Components = 'Components/',
    Templates = 'Templates/',
    Iconography = 'Iconography/',
    // Example of new root:
    // Layout = "Layout/",
}

export enum LibPath {
    Cards = Root.Components + 'Cards/',
    Containers = Root.Components + 'Containers/',
    Form = Root.Components + 'Form/',
    Navigation = Root.Components + 'Navigation/',
    Overlay = Root.Components + 'Overlay/',
    Disclosure = Root.Components + 'Disclosure/',
    Feedback = Root.Components + 'Feedback/',
    // Example of new sub-directory
    // Headers = Root.Layout + "Headers/",
}
