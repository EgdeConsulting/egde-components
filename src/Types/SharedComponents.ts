import {
	Dispatch,
	MutableRefObject,
	SetStateAction,
	ReactElement,
} from 'react';

type Direction = 'row' | 'column';

enum IconTestID {
	AlertWarning = 'alertWarningIcon',
	AlertInfo = 'alertInfoIcon',
	AlertError = 'alertErrorIcon',
	AlertSuccess = 'alertSuccessIcon',
	AlertDialog = 'alertDialogIcon',
	CardTitle = 'cardTitleIcon',
	CardAction = 'cardActionIcon',
	Modal = 'modalIcon',
}

enum ModalIconType {
	Check = 'check',
	Exclamation = 'exclamation',
}

enum AlertStatus {
	Warning = 'warning',
	Info = 'info',
	Error = 'error',
	Success = 'success',
}
enum AlertDialogIconType {
	Warning = 'warning',
}

type PageContainerProps = {
	title: string;
	titleAlignment?: 'left' | 'right' | 'center';
	titleTextColor?: string;
	subtitle?: string;
	subtitleAlignment?: 'left' | 'right' | 'center';
	subtitleTextColor?: string;
	children: JSX.Element[] | JSX.Element;
	containerPadding?: string | (string | number)[];
	containerMargin?: string | (string | number)[];
	containerSideMargins?: string | (string | number)[];
	contentPadding?: string | (string | number)[];
	contentMargin?: string | (string | number)[];
	subtitlePadding?: string | (string | number)[];
	titlePadding?: string | (string | number)[];
	titleMargin?: string | (string | number)[];
	borderColor?: string;
	borderWidth?: string | (string | number)[];
	contentBorderColor?: string;
	contentBorderWidth?: string;
	titleBadge?: { badgeColor: string; textColor: string; label: string };
	backgroundColor?: string;
	contentColor?: string;
	titlesColor?: string;
	height?: string | (string | number)[];
	width?: string | (string | number)[];
	containerMinWidth?: string | (string | number)[];
	containerMaxWidth?: string | (string | number)[];
	shouldContentFillContainer?: boolean;
	/** Use a single or array of 'ch' values, such as '60ch' or ['50ch', '60ch', '80ch'] for responsive breakpoints */
	containerSize?: string | string[];
};

interface BaseInputProps extends InputProps {
	children: JSX.Element[] | JSX.Element;
}

type InputProps = {
	label?: string;
	labelTextMargin?: string;
	invalidText?: string;
	captionText?: string;
	captionTextMargin?: string;
	margin?: string;
	padding?: string;
	width?: string;
	minWidth?: string;
	maxWidth?: string;
};
interface InputComponentProps extends InputProps {
	isDisabled?: boolean;
	backgroundColor?: string;
	autoFocus?: boolean;
}
interface TextInputProps extends InputComponentProps {
	placeholder?: string;
	value: string;
	onChange: Dispatch<SetStateAction<string>>;
	characterLimit?: number;
	textAlign?: 'right' | 'center';
}

interface NumberInputProps extends InputComponentProps {
	placeholder?: string;
	allowNegative?: boolean;
	value: string;
	onChange: Dispatch<SetStateAction<string>>;
	maxValue?: string;
	maxValueLength?: number;
	isCurrency?: boolean;
	currencySpacer?: string;
	textAlign?: 'right' | 'center';
}

type CardProps = {
	title: string;
	body: string;
	onClick?: () => void;
	titleIcon?: ReactElement;
	actionIcon?: ReactElement;
	cardHeight?: string;
	cardWidth?: string;
	titleSize?: string;
};

type ModalProps = {
	isModalOpen: boolean;
	isCentered: boolean;
	iconVariant?: ModalIconType;
	title: string;
	subtitle?: string;
	children?: JSX.Element[] | JSX.Element;
	buttons: ModalButtons;
	hasCloseButton?: boolean;
	onClose: () => void;
	errorMessage?: string;
	width?: string;
};

type ModalButtons = {
	primary: ModalButton;
	secondary: ModalButton;
	alignment: 'left' | 'right';
};

type ModalButton = {
	label: string;
	variant: string;
	onClick?: () => void;
};

interface RadioProps extends InputComponentProps {
	direction?: Direction;
	options: string[];
	value: number;
	onChange: Dispatch<SetStateAction<number>>;
}

interface CheckboxProps extends InputComponentProps {
	direction?: Direction;
	options: string[];
	value: number[];
	onChange: Dispatch<SetStateAction<number[]>>;
}

type AccordionProps = {
	allowMultiple?: boolean;
	items: AccordionItem[];
};

type AccordionItem = {
	title: string;
	content: JSX.Element[] | JSX.Element;
};

type TabProps = {
	tabs: Tab[];
	centerTabs?: boolean;
};

type Tab = {
	title: string;
	children: JSX.Element[] | JSX.Element;
};

type StepperProps = {
	activeStep: number;
	setActiveStep: Dispatch<SetStateAction<number>>;
	stepsContent: StepContent[];
	buttonTexts: StepperButtonTexts;
	finalStep: finalStep;
	buttonAlignment?: 'left' | 'center' | 'right';
	buttonMargin?: string;
	clickableSteps?: boolean;
	completedSteps?: number[];
	onFinalStep: () => void;
	onFinalize: () => void;
	isCompleteButtonDisabled?: boolean;
	buttonCaption?: { text: string; color: string };
};

type StepContent = {
	label: string;
	children: JSX.Element[] | JSX.Element;
};
type StepperButtonTexts = {
	previous: string;
	next: string;
	complete: string;
};
type finalStep = {
	finalText: string;
	finalButtonText: string;
};

type AlertProps = {
	status: AlertStatus;
	description: string;
	onClose: () => void;
	linkText?: string;
	linkHref?: string;
};

type AlertDialogProps = {
	iconVariant?: AlertDialogIconType;
	heading: string;
	text: string;
	isOpen: boolean;
	cancelRef: MutableRefObject<undefined>;
	onClose: () => void;
	buttons: AlertDialogButton[];
};

type AlertDialogButton = {
	label: string;
	variant: string;
	onClick: () => void;
};

interface BaseSelectProps extends InputComponentProps {
	placeholder?: string;
	options: SelectOption[];
	filterText?: string;
}

interface SelectProps extends BaseSelectProps {
	onChange: Dispatch<SetStateAction<SelectOption | undefined>>;
	value?: SelectOption;
}

interface MultiSelectProps extends BaseSelectProps {
	submitText?: string;
	cancelText?: string;
	onChange: Dispatch<SetStateAction<SelectOption[]>>;
	value: SelectOption[];
}

type SelectOption = {
	label: string;
	value: number;
};

interface DatePickerProps extends InputComponentProps {
	placeholder?: string;
	onChange: React.Dispatch<React.SetStateAction<Date | undefined>>;
	value?: Date;
	maxDate?: Date;
	minDate?: Date;
	shouldCloseOnSelect?: boolean;
}
interface RichTextAreaProps extends TextInputProps {
	toolbarOptions?: {
		bold?: boolean;
		italic?: boolean;
		underline?: boolean;
		unorderedList?: boolean;
	};
}
interface FileUploadProps extends InputProps {
	uploadLabel?: string;
	dropLabel?: string;
	maxFileSizeLabel?: string;
	maxUploadSizeLabel?: string;
	maxFileSize: number;
	maxUploadSize: number;
	disabled?: boolean;
	showIcon?: boolean;
	size?: string;
	variant?: string;
}

export type {
	PageContainerProps,
	BaseInputProps,
	TextInputProps,
	NumberInputProps,
	CardProps,
	ModalProps,
	ModalButtons,
	RadioProps,
	AccordionProps,
	AccordionItem,
	CheckboxProps,
	TabProps,
	StepperProps,
	AlertProps,
	AlertDialogProps,
	SelectProps,
	MultiSelectProps,
	SelectOption,
	AlertDialogButton,
	DatePickerProps,
	RichTextAreaProps,
	FileUploadProps,
};

export { IconTestID, AlertStatus, AlertDialogIconType, ModalIconType };
