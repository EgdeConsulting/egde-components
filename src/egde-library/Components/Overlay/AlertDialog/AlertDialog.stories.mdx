import { Meta, Story, Canvas } from '@storybook/addon-docs';
import { LibPath, DocsPageHeader, DocsPageBodyComponents } from 'egde-library';
import { AlertDialog } from 'egde-components';
import { useState } from 'react';
import { Button } from '@chakra-ui/react';

export const Template = (args) => {
    const [isOpen, setIsOpen] = useState(false);
    const handleClose = () => setIsOpen(false);
    const buttons = [
        {
            label: 'OK',
            onClick: () => {
                setIsOpen(false);
            },
            variant: 'primary',
        },
    ];
    return (
        <div>
            <Button onClick={() => setIsOpen(true)}> Open Alert Dialog </Button>
            <AlertDialog
                {...args}
                onClose={handleClose}
                isOpen={isOpen}
                buttons={buttons}
            />
        </div>
    );
};

export const componentName = 'AlertDialog';

<Meta
    title={LibPath.Overlay + componentName}
    component={AlertDialog}
    args={{
        iconVariant: 'information',
        heading: 'Title',
        text: 'Content',
        isOpen: true,
        cancelRef: React.createRef(),
        onClose: () => {},
        buttons: { label: 'OK', onClick: () => {}, variant: 'primary' },
    }}
/>

<DocsPageHeader
    title={componentName}
    description=" This component is used to create an overlay that displays an alert. Pressing the button opens the AlertDialog."
    addImportantNote={false}
    variant="stable"
/>

<Canvas withToolbar>
    <Story name="Base">{Template.bind({})}</Story>
</Canvas>

<DocsPageBodyComponents addStories={true} />

### Warning Alert

<Canvas withToolbar>
    <Story
        name="Warning Alert"
        args={{
            iconVariant: 'warning',
            heading: 'Something went wrong!',
            text: 'Please try again',
        }}
    >
        {Template.bind({})}
    </Story>
</Canvas>
