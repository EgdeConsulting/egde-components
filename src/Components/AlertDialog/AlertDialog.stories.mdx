import {
    Meta,
    Story,
    Props,
    Canvas,
    Preview,
} from '@storybook/addon-docs/blocks';
import { AlertDialog } from './AlertDialog';
import { ComponentStability } from 'Icons';
import { LibPath } from 'Types';
import { useState } from 'react';
import { Button } from '@chakra-ui/react';

export const Template = (args) => {
    const [isOpen, setIsOpen] = useState(false);
    const handleClose = () => setIsOpen(false);
    const buttons = [
        {
            label: 'OK',
            onClick: () => {
                setIsOpen(false);
            },
            variant: 'primary',
        },
    ];
    return (
        <div>
            <Button onClick={() => setIsOpen(true)}> Open Alert Dialog </Button>
            <AlertDialog
                {...args}
                onClose={handleClose}
                isOpen={isOpen}
                buttons={buttons}
            />
        </div>
    );
};

<Meta
    title={LibPath.Overlay + 'AlertDialog'}
    component={AlertDialog}
    args={{
        iconVariant: 'information',
        heading: 'Heading',
        text: 'Alert Dialog Text',
        isOpen: true,
        cancelRef: React.createRef(),
        onClose: () => {},
        buttons: { label: 'OK', onClick: () => {}, variant: 'primary' },
    }}
/>

# Alert Dialog

<ComponentStability variant="stable" />

<hr />

This component is used to display a popup-box which displays important information
whom the user must see.

<Preview withToolbar>
    <Story name="Base">{Template.bind({})}</Story>
</Preview>

Change some of the props and see how the `Base` component changes.

<Props story="Base" />

# Stories

<hr />

Below are shown different implementations of the component in regards to varying
use cases.

<br />

<Canvas withToolbar>
    <Story
        name="Warning Alert"
        args={{
            iconVariant: 'warning',
            heading: 'Something went wrong!',
            text: 'Please try again',
        }}
    >
        {Template.bind({})}
    </Story>
</Canvas>
