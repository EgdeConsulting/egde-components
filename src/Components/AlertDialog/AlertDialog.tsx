import {
    AlertDialog as ChakraAlertDialog,
    AlertDialogBody,
    AlertDialogFooter,
    AlertDialogContent,
    AlertDialogOverlay,
    Button,
    AlertDialogHeader,
    Icon,
    Flex,
    Heading,
    Text,
} from '@chakra-ui/react';
import React, { ReactElement } from 'react';
import { AlertDialogProps, IconTestID, AlertDialogIconType } from 'Types';
import { Warning } from 'Icons';

export const AlertDialog = (props: AlertDialogProps): ReactElement => {
    const {
        iconVariant,
        heading,
        text,
        isOpen,
        cancelRef,
        onClose,
        buttons } = props;

    const iconStyle = {
        color: 'red',
        marginRight: '8px',
        size: '24px',
        testID: IconTestID.AlertDialog,
        ...(iconVariant === AlertDialogIconType.Warning && {
            icon: Warning,
        }),
    };
    return (
        <ChakraAlertDialog
            isOpen={isOpen}
            leastDestructiveRef={cancelRef.current}
            onClose={onClose}
        >
            <AlertDialogOverlay>
                <AlertDialogContent>
                    <AlertDialogHeader>
                        <Flex width='100%' direction='row'>
                            {iconVariant && (
                                <Icon
                                    as={Warning}
                                    color={iconStyle.color}
                                    marginRight={iconStyle.marginRight}
                                    style={{ fontSize: iconStyle.size }}
                                    data-testid={iconStyle.testID}
                                />
                            )}
                            <Heading size='md'>{heading}</Heading>
                        </Flex>
                    </AlertDialogHeader>

                    <AlertDialogBody>
                        <Text>{text}</Text>
                    </AlertDialogBody>

                    <AlertDialogFooter>
                        {buttons.map(({ label, variant, onClick }, index) => (
                            <Button
                                variant={variant}
                                onClick={onClick}
                                key={`adfbtn${index}`}
                            >
                                {label}
                            </Button>
                        ))}
                    </AlertDialogFooter>
                </AlertDialogContent>
            </AlertDialogOverlay>
        </ChakraAlertDialog>
    );
};
