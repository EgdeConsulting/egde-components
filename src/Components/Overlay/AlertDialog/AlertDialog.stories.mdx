import { Meta, Story, Canvas } from '@storybook/addon-docs';
import { DocsPageHeader, DocsPageBody } from 'Docs';
import { LibPath } from 'Types';
import { AlertDialog } from 'Components';
import { useState } from 'react';
import { Button } from '@chakra-ui/react';

export const Template = (args) => {
    const [isOpen, setIsOpen] = useState(false);
    const handleClose = () => setIsOpen(false);
    const buttons = [
        {
            label: 'OK',
            onClick: () => {
                setIsOpen(false);
            },
            variant: 'primary',
        },
    ];
    return (
        <div>
            <Button onClick={() => setIsOpen(true)}> Open Alert Dialog </Button>
            <AlertDialog
                {...args}
                onClose={handleClose}
                isOpen={isOpen}
                buttons={buttons}
            />
        </div>
    );
};

export const componentName = 'AlertDialog';

<Meta
    title={LibPath.Overlay + componentName}
    component={AlertDialog}
    args={{
        iconVariant: 'information',
        heading: 'Title',
        text: 'Content',
        isOpen: true,
        cancelRef: React.createRef(),
        onClose: () => {},
        buttons: { label: 'OK', onClick: () => {}, variant: 'primary' },
    }}
/>

<DocsPageHeader
    title={componentName}
    description=" This component is used to create an overlay that displays an alert. Pressing the button opens the AlertDialog."
    addImportantNote={false}
    variant="stable"
/>

<Canvas withToolbar>
    <Story name="Base">{Template.bind({})}</Story>
</Canvas>

<DocsPageBody addStories={true} />

### Warning Alert

<Canvas withToolbar>
    <Story
        name="Warning Alert"
        args={{
            iconVariant: 'warning',
            heading: 'Something went wrong!',
            text: 'Please try again',
        }}
    >
        {Template.bind({})}
    </Story>
</Canvas>
