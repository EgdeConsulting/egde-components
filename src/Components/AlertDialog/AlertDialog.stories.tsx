import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { AlertDialog } from './AlertDialog';
import { LibPath } from 'Types';

export default {
  title: LibPath.Overlay + 'AlertDialog' || 'undefined',
  component: AlertDialog,
  argTypes: {
    iconVariant: { defaultValue: '' },
        heading: { defaultValue: 'Heading' },
        text: { defaultValue: 'Text' },
        isOpen: { defaultValue: true },
        cancelRef: { defaultValue: React.createRef() },
        onClose: { action: 'onClose' },
        buttons: { defaultValue: 
        {
          lablel: 'Cancel',
          variant: 'primary',
          onclick: () => {},
        } },
  },
} as ComponentMeta<typeof AlertDialog>;

const Template: ComponentStory<typeof AlertDialog> = (args) => 
<AlertDialog {...args} />;


export const VerticalStepperTest = Template.bind({});
