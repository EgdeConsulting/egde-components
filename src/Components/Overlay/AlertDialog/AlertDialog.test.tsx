import React, { MutableRefObject } from 'react';
import '@testing-library/jest-dom';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ReactElement, useRef, useState } from 'react';
import { AlertDialogButton, AlertDialogIconType, IconTestID } from 'Types';
import { AlertDialog } from './AlertDialog';
import ReactDOM from 'react-dom';

const ALERT_DIALOG_TESTER_ICON_ID = IconTestID.AlertDialog;
const ALERT_DIALOG_TESTER_HEADING = 'This is the heading';
const ALERT_DIALOG_TESTER_TEXT = 'This is the text';
const ALERT_DIALOG_TESTER_BUTTON_TEXT = 'Confirm button';

const AlertDialogTester = (): ReactElement =>{
    const [isOpen, setIsOpen] = useState<boolean>(false);
    const cancelRef = useRef();
    const warningAlertButtons: AlertDialogButton[] = [
        {
            label: ALERT_DIALOG_TESTER_BUTTON_TEXT,
            variant: 'primary',
            onClick: () => setIsOpen(false),
        },
    ];

    return (
        <>
            <button type="button" onClick={() => setIsOpen(true)}>
                Open
            </button>
            <AlertDialog
                iconVariant={AlertDialogIconType.Warning}
                heading={ALERT_DIALOG_TESTER_HEADING}
                text={ALERT_DIALOG_TESTER_TEXT}
                isOpen={isOpen}
                cancelRef={cancelRef}
                onClose={() => setIsOpen(false)}
                buttons={warningAlertButtons}
            />
        </>
    );
}
describe('Alert component test', () => {
    it('matches snapshot', () => {
        const tree = render(<AlertDialogTester />);
        expect(tree).toMatchSnapshot();
    });
    it('renders without chrashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<AlertDialogTester/>, div);
    })
    it.skip('assert content is null, alert dialog opens when button clicked, and alert dialog content is visible', async () => {
        render(<AlertDialogTester />);
        expect(screen.queryByText(ALERT_DIALOG_TESTER_TEXT)).toBeNull();
        expect(screen.queryByText(ALERT_DIALOG_TESTER_BUTTON_TEXT)).toBeNull();
        const openAlertDialogButton = screen.getByRole('button');
        userEvent.click(openAlertDialogButton);
        const icon = await screen.findByTestId(ALERT_DIALOG_TESTER_ICON_ID);
        await waitFor(() => expect(icon).toBeVisible());
        expect(
            await screen.findByText(ALERT_DIALOG_TESTER_HEADING),
        ).toBeVisible();
        expect(await screen.findByText(ALERT_DIALOG_TESTER_TEXT)).toBeVisible();
        expect(
            await screen.findByRole('button', {
                name: ALERT_DIALOG_TESTER_BUTTON_TEXT,
            }),
        ).toBeVisible();
    });

    it('assert button in alert dialog exists, and clicking it closes the alert dialog', async () => {
        render(<AlertDialogTester />);
        const openAlertDialogButton = screen.getByRole('button');
        userEvent.click(openAlertDialogButton);
        const heading = await screen.findByText(ALERT_DIALOG_TESTER_HEADING);
        const button = await screen.findByRole('button', {
            name: ALERT_DIALOG_TESTER_BUTTON_TEXT,
        });
        if (heading && button) {
            userEvent.click(button);
            await waitFor(() => expect(heading).not.toBeVisible());
        }
    });
});
