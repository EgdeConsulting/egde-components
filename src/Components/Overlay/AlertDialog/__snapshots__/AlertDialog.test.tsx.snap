// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Alert should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
    />
    <div
      class="chakra-portal"
    >
      <div
        class="chakra-modal__overlay css-13d1tkw"
        style="opacity: 0;"
      >
        <div
          aria-hidden="true"
          data-aria-hidden="true"
          data-focus-guard="true"
          style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
          tabindex="0"
        />
        <div
          aria-hidden="true"
          data-aria-hidden="true"
          data-focus-guard="true"
          style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
          tabindex="1"
        />
        <div
          data-focus-lock-disabled="false"
        >
          <div
            class="chakra-modal__content-container css-k2m3mz"
            tabindex="-1"
          >
            <section
              aria-describedby="chakra-modal--body-1"
              aria-labelledby="chakra-modal--header-1"
              aria-modal="true"
              class="chakra-modal__content css-6yiygt"
              id="chakra-modal-1"
              role="alertdialog"
              style="opacity: 0; transform: scale(0.95) translateZ(0);"
              tabindex="-1"
            >
              <header
                class="chakra-modal__header css-72fd9l"
                id="chakra-modal--header-1"
              >
                <div
                  class="css-1ik4laa"
                >
                  <svg
                    fill="#CC0000"
                    height="24px"
                    viewBox="0 0 24 24"
                    width="24px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
                    />
                  </svg>
                  <h2
                    class="chakra-heading css-0"
                  >
                    Heading
                  </h2>
                </div>
              </header>
              <div
                class="chakra-modal__body css-0"
                id="chakra-modal--body-1"
              >
                <p
                  class="chakra-text css-0"
                >
                  Alert Dialog Text
                </p>
              </div>
              <footer
                class="chakra-modal__footer css-192qrng"
              >
                <button
                  class="chakra-button css-5c2176"
                  type="button"
                >
                  Cancel
                </button>
              </footer>
            </section>
          </div>
        </div>
        <div
          aria-hidden="true"
          data-aria-hidden="true"
          data-focus-guard="true"
          style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
          tabindex="0"
        />
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
    data-aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
